File: code/Java/Chapter12/12-4/JumpGame.java
public class JumpGame {
    public static void main(String[] args) {
        int[] stones = {2, 3, 1, 1, 4};
        System.out.println(canJump(stones));
    }

    public static boolean canJump(int[] stones) {
        boolean[] canJump = new boolean[stones.length];
        canJump[0] = true;

        for(int j = 1; j < stones.length; j++) {
            canJump[j] = false;
            for(int i = 0; i < j; i++) {
                if(canJump[i] && i + stones[i] >= j) {
                    canJump[j] = true;
                    break;
                }
            }
        }
        return canJump[stones.length - 1];
    }
}

File: code/Python/Chapter12/12-4/jump_game.py
def can_jump(stones):
    can_jump = [True] + [False] * (len(stones) - 1)

    for j in range(1, len(stones)):
        for i in range(j):
            if can_jump[i] and i + stones[i] >= j:
                can_jump[j] = True
                break
    return can_jump[len(stones) - 1]

stones = [2, 3, 1, 1, 4]
print(can_jump(stones))